_posts/2025-10-23-mmaler-blogpost-2-quarkus-runtime-and-framework-for-cloud-native-java.adoc

```diff
@@ -0,0 +1,152 @@
++
+image::3-BuildTimeP-Benefits-JVM.png[alt="Benefits of Quarkus build-time processing on the JVM", width=100%]
++
+image::4-BuildTimeP-Benefits-native.png[alt="Benefits of Quarkus native image compared to JVM", width=100%]
++
+image::5-JVM-Snapshotting-Native.png[alt="JVM snapshotting vs native images in Quarkus", width=100%]
+
+* *Extensibility*: Quarkus exposes extension points for everything from startup hooks to request filters.
+Over 800 extensions allow seamless integration with modern technologies such as Kafka, OpenTelemetry, and OpenID Connect.
+
+These extensions integrate with Quarkus and participate in its build-time and runtime lifecycle, making them first-class citizens of the Quarkus platform.
+
+=== Simplified developer experience
+
+Frameworks succeed when they reduce complexity without sacrificing power.
+Quarkus does exactly that:
+
+* Preconfigures popular libraries with sensible defaults.
+* Offers unified configuration and developer tooling.
+* Provides instant feedback with live reload and continuous testing.
+
+This makes Quarkus both powerful and approachable.
+
+You can start with a simple REST endpoint and scale it into a production-grade service without changing your development model.
+
+image::6-Frictionless.png[alt="Frictionless developer experience: live reload, dev services, continuous testing", width=100%]
+
+These features reflect Quarkus’s dual role: it behaves like a framework during development, offering structure, defaults, and conventions, and it acts as a runtime during execution, delivering performance, efficiency, and operational consistency.
+
+=== Performance that matters
+
+As described in the link:https://cacm.acm.org/practice/application-frameworks/[ACM article on application frameworks], Google's internal framework design emphasized fast startup and resilience against cascading failures.
+Quarkus was built with these principles in mind.
+
+Native image generation with GraalVM enables cold starts in milliseconds and a minimal memory footprint, allowing cost-effective deployment in containers and serverless environments.
+
+Quarkus also includes structured health checks, metrics, and tracing, thereby aligning with production standards from day one.
+
+image::7-success-stories.png[alt="Real-world success stories using Quarkus", width=100%]
+
+
+=== Secure by default
+
+Quarkus separates application logic from infrastructure concerns.
+By structuring the request lifecycle with standardized handlers such as exception mappers, filters, and interceptors, it supports strong security practices:
+
+* Secure headers and CSRF protection out of the box.
+* Native support for role-based access control.
+* Seamless integration with OpenID Connect providers.
+
+Just as Google’s frameworks reduced vulnerabilities through safe APIs and typed response models, Quarkus applies similar principles by using conventions and standardized components to promote secure defaults and trusted integrations.
```

Comment:
Similar comments here, Quarkus is not secure by default

Reply:
Same here, I don't see why you're trying to tie us to Google.